ASFAchievementManager/AchievementHandler.cs

```diff
@@ -20,23 +20,21 @@ public sealed class AchievementHandler : ClientMsgHandler {
 			switch (packetMsg.MsgType) {
 				case EMsg.ClientGetUserStatsResponse:
 					ClientMsgProtobuf<CMsgClientGetUserStatsResponse> getAchievementsResponse = new(packetMsg);
-					Client.PostCallback(new GetAchievementsCallback(packetMsg.TargetJobID, getAchievementsResponse.Body));
-
+					Client.PostCallback(new GetAchievementsCallback(packetMsg.TargetJobID, getAchievementsResponse.Body, (msg) => msg.eresult));
 					break;
 				case EMsg.ClientStoreUserStatsResponse:
 					ClientMsgProtobuf<CMsgClientStoreUserStatsResponse> setAchievementsResponse = new(packetMsg);
-					Client.PostCallback(new SetAchievementsCallback(packetMsg.TargetJobID, setAchievementsResponse.Body));
-
+					Client.PostCallback(new SetAchievementsCallback(packetMsg.TargetJobID, setAchievementsResponse.Body, (msg) => msg.eresult));
 					break;
 			}
 
 		}
 
-		internal sealed class GetAchievementsCallback : CallbackMsg {
-			internal readonly CMsgClientGetUserStatsResponse Response;
+		internal abstract class AchievementsCallBack<T> : CallbackMsg {
+			internal readonly T Response;
 			internal readonly bool Success;
 
-			internal GetAchievementsCallback(JobID jobID, CMsgClientGetUserStatsResponse msg) {
+			internal AchievementsCallBack(JobID jobID, T msg, Func<T, int> eresult, string error) {
```

Comment:
proper name for the parameter would probably be `resultGetter` or `resultExtractor` since it is not the "eresult" but rather a function extracting it from the protobuf response.